include/app/log.hpp

```diff
@@ -92,3 +90,64 @@ class Log {
 #define LOG_WARNING(...)  Log::warning  ("[", __FILENAME__, ":", __LINE__, "] ", __VA_ARGS__)
 #define LOG_ERROR(...)    Log::error    ("[", __FILENAME__, ":", __LINE__, "] ", __VA_ARGS__)
 #define LOG_CRITICAL(...) Log::critical ("[", __FILENAME__, ":", __LINE__, "] ", __VA_ARGS__)
+
+template <typename ... Args>
+void Log::message(Level level, const Args &... args) {
+    // TODO: output to log file
+    // TODO: choose message format
+    // TODO: add filtering by level
+    std::stringstream ss;
+    ss << "[" << SystemClock::getTime() << "] " << level << ' ';
+    (print<Args>(ss, args), ...);
+    ss << std::endl;
+    if (_config.write_to_console) {
+        std::cerr << ss.str();
+    }
+    if (log_out_fs.is_open()) {
+        log_out_fs << ss.str();
+    }
```

Comment:
std::stringstream initialization takes very long, i think it's better to make it static and reuse or write to both files separately

Reply:
Made it a static member of Log class